pkg/generate/app/cmd/newapp.go

```diff
@@ -586,7 +586,18 @@ func (c *AppConfig) validate() (app.Environment, app.Environment, error) {
 		return nil
 	})
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
+	}
+	buildEnv, err := app.ParseAndCombineEnvironment(c.BuildEnvironment, c.BuildEnvironmentFiles, c.In, func(key, file string) error {
```

Comment:
GetBuildEnvironment calls c.validate() which returns the data there, which is called by the pipeline builder on line 285.  I could probably refactor the code there to to remove passing environment to GetBuildEnvironment since it's not used anymore.